src/persian_content/mod.rs

```diff
@@ -34,7 +26,6 @@ pub trait PersianContent: AsRef<str> {
         let string = self
             .as_ref()
             .chars()
-            .into_iter()
             .filter(|c| c.is_alphabetic())
             .collect::<String>();
```

Comment:
Instead of the whole logic in this function, we can use the following code to avoid the allocation.
But generally I'm not sure if this is generally faster or not!
```
let (persian_chars, len) = self
            .as_ref()
            .chars()
            .filter(|c| c.is_alphabetic()) // First filter out non-alphabetic chars
            .fold((0, 0), |(persian_chars, len), c| {
                if HAS_PERSIAN_CHAR.contains(&c) {
                    (persian_chars + 1, len + 1)
                } else {
                    (persian_chars, len + 1)
                }
            });
```